src/PostgREST/App.hs

```diff
@@ -105,6 +105,19 @@ app conf reqBody req =
             , (hLocation, "/postgrest/users?id=eq." <> cs (userId u))
             ] ""
 
+    (["postgrest", "users"], "PUT") -> do
+      let user = decode reqBody :: Maybe AuthUser
+
+      case user of
+        Nothing -> return $ responseLBS status400 [jsonH] $
+          encode . object $ [("message", String "Failed to parse user.")]
+        Just u -> do
+           _ <- updateUser (cs $ userId u) (cs $ userPass u) (cs $ userRole u)
+           return $ responseLBS status200
+            [ jsonH
+            , (hLocation, "/postgrest/users?id=eq." <> cs (userId u))
```

Comment:
If the PUT applies to the user resource itself then we won't need to return a location. Also the location header with a 200 response is unconventional.